---
title: "RNAseq analysis for LVNC project-cardiomyocytes"
output: word_document
date: "3-11-2024"
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r, eval=TRUE, echo= TRUE}
setwd("Z:/ResearchHome/Groups/sapkogrp/projects//RNAseq/common/Novogene_Romina_RNAseq/")
```

## Project description (Notes from Romina)
The dataset I am sending corresponds to RNAseq samples from cardiac fibroblasts (CF) and I have 4 cells lines (isogenic cell line (ISO), ROBO1, ROBO2, and SLIT2 knockouts). In addition, the cells were treated with 0 or 0.3uM doxorubicin (DOX). 
The goal is to address:
1- Differential expression in the ISO vs KOs cells (not treated with DOX)
2- Differential expression after treatment with DOX for each cell line (and compare to ISO)

Let me know how I can help and if you have any questions.


# Loading libraries and setting location paths

```{r, eval=TRUE, echo= TRUE}

library(ggplot2)
library(ggfortify)
library(ggrepel)
library(dplyr)
library(DESeq2)
library(ggplot2)
library("knitr")
library("kableExtra")
```

# Preparing count data
```{r, eval=TRUE, echo= TRUE}
counts<-read.table(file='Z:/ResearchHome/Groups/sapkogrp/projects//RNAseq/common/Novogene_Romina_RNAseq/SAPKO-821633-UNSTRANDED_RSEM_gene_count.2024-03-06_01-10-53.txt', header=TRUE)
# > dim(counts)
# [1] 60754    52


counts$geneSymbol[duplicated(counts$geneSymbol)] <- paste0(counts$geneSymbol, "_", counts$geneID)[duplicated(counts$geneSymbol)]

counts = dplyr::select(counts, -c(geneID, bioType, annotationLevel))
counts = counts[rowSums(counts[,-1])>20,]

rownames(counts)<-counts[,1]
counts2<-counts[,-1]


properties <- read.table(text="Samples	Project	samples.Pr	Cell_type	Dox_treatment
ISO	Pr1	Pr1_ISO	ISO_CF	No
ISO_DOX	Pr1	Pr1_ISO_DOX	ISO_CF	Yes
ROBO1	Pr1	Pr1_ROBO1	ROBO1_KO_CF	No
ROBO1_DOX	Pr1	Pr1_ROBO1_DOX	ROBO1_KO_CF	Yes
ROBO2	Pr1	Pr1_ROBO2	ROBO2_KO_CF	No
ROBO2_DOX	Pr1	Pr1_ROBO2_DOX	ROBO2_KO_CF	Yes
SLIT2	Pr1	Pr1_SLIT2	SLIT2_KO_CF	No
SLIT2_DOX	Pr1	Pr1_SLIT2_DOX	SLIT2_KO_CF	Yes", header = T, sep = "\t")


```



```{r, eval=TRUE, echo= TRUE}
## data preparaton
counts.matrix_ROBI<-as.matrix(counts2)
mode(counts.matrix_ROBI)<-"integer"

counts.matrix_ROBI <- counts.matrix_ROBI[,grepl("Pr1", colnames(counts.matrix_ROBI))]
colnames(counts.matrix_ROBI) <- gsub("Pr1_", "", colnames(counts.matrix_ROBI))

properties$Cell_type=as.factor(properties$Cell_type)
properties$Group=as.factor(properties$Dox_treatment)
properties$sample_ID <- properties$Samples
properties <- properties[, c("sample_ID", "Group", "Cell_type")] # remove Cell_type
counts.matrix_ROBI <- counts.matrix_ROBI[,match(properties$sample_ID, colnames(counts.matrix_ROBI))]
colnames(counts.matrix_ROBI) == properties$sample_ID


## Create a DESeqDataSet (dds_ROBIS) from the count matrix and phenotype information.
dds_ROBIS<-DESeqDataSetFromMatrix(countData = counts.matrix_ROBI,colData = properties, design=~1)

## Perform variance-stabilizing transformation (vst) on the DESeqDataSet
rld_ROBIS<-vst(dds_ROBIS,blind = FALSE)
mat_ROBIS<-assay(rld_ROBIS)

## Performs principal component analysis (PCA) on the transformed data.
pcaData<-plotPCA(rld_ROBIS, intgroup=c("Group", "Cell_type"), ntop=100, returnData=TRUE)
percentVar<-round(100 * attr(pcaData, "percentVar"))

## Plot PCA
g <- ggplot(pcaData, aes(PC1, PC2))
g <- g + geom_point(size = 5, aes(fill = Group, col = Group, shape = Cell_type)) +
  theme_bw() +
  theme(
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.text = element_text(face = "bold", color = "black"),
    panel.border = element_blank(),
    axis.line = element_line(colour = "black"),
    axis.title = element_text(face = "bold", color = "black"),  # Corrected line
    text = element_text(color = "black", face = "bold")
  )
# Save the plot as a TIFF file
tiff(
  file = "Pr1_top_100_genes.tiff",
  width = 16,
  height = 14,
  units = 'cm',
  res = 300,
  compression = 'lzw'
)

# Print the plot
print(g)

# Close the TIFF device
dev.off()

## Select the top 500 variable genes based on row variances. Create a heatmap
## (phm) using pheatmap with clustering of rows and columns.
library(pheatmap)
properties2<-properties
rownames(properties2)<-properties[,1]
properties2<-subset(properties2, select=-c(sample_ID))
myMatrix<-SummarizedExperiment::assay(rld_ROBIS)
topVarGenes<-head(order(rowVars(myMatrix),decreasing = TRUE), 500)
myMatrixforheatmap<-myMatrix[topVarGenes,]
phm<-pheatmap(myMatrixforheatmap, scale="row", cluster_rows = TRUE, cluster_cols = TRUE, annotation_col = properties2, show_colnames = TRUE, show_rownames = FALSE)
tiff(file="Pr1_top_500_genes.tiff",width = 18, height = 32, units = 'cm', res = 300, compression = 'lzw')
phm
dev.off()
```


## 1.The first question is to see differences between control and cardiomyopathy group. 

Here, we will conduct differential expression analysis between "Cardiomyopathy" and "Control" groups, filter and categorize significant genes, and create a volcano plot for visualization. The volcano plot highlights genes with significant changes in expression, facilitating the identification of differentially expressed genes.The design in this context includes the factors Status (indicating whether a sample is from the Cardiomyopathy or Control group).

Since there could be inherent differences in gene expression levels between different cell types (like some cell types naturally express certain genes more than others), treating Cell_type as a fixed effect might not fully capture these differences.Thus, we might miss capturing the baseline variability in gene expression levels between different cell types. This could lead to biased or inaccurate results when analyzing the effect of Group on gene expression.However, including a random effect for Cell_type helps to control for these inherent differences in gene expression levels between different cell types.In addition, none of the negative binomial GLM-based packages (e.g. edgeR & DESeq2) support models with random effects.(https://support.bioconductor.org/p/115282/)

Dream: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8055218/; https://bioconductor.riken.jp/packages/devel/bioc/vignettes/variancePartition/inst/doc/dream.html

This approach ensures that any observed effect of Group on gene expression is not confounded by differences in cell types. It helps to separate the effects of the experimental treatment (Group) from the inherent differences in gene expression between different cell types.

```{r, eval=TRUE, echo= TRUE}
# # Create a DESeqDataSet (ddset) from the count matrix (counts.matrix_ROBI) and
# # phenotype information (properties2).
# ddset = DESeqDataSetFromMatrix(
#     countData = counts.matrix_ROBI, # raw counts
#     colData = properties2, # phenotype data
#     design = ~Group + (1 | Cell_type)
# )
# 
# deseq_object = DESeq(ddset)
# 

####################
## Dream analysis ##
####################
library("variancePartition")
library("edgeR")
library(limma)
library("BiocParallel")
# https://www.bioconductor.org/packages/devel/workflows/vignettes/RNAseq123/inst/doc/limmaWorkflow.html
## metadata
info <-  properties2[, c("Group", "Cell_type")]
# info$Group <- factor(info$Group, levels= c("Yes", "No"))

# Create a DGEList object and normalize the data:
y <- DGEList(counts=counts.matrix_ROBI)
y <- calcNormFactors(y)

# Design matrix and blocking factor:
design <- model.matrix(~ Group, data=info)
block <- info$Cell_type

# Estimate the dispersion:
y <- estimateDisp(y, design)

# Convert the DGEList to a voom object:
v <- voom(y, design)

# Estimate the correlation: The duplicateCorrelation function estimates the
# correlation between repeated measures or related samples. In our case, it is
# used to account for the random effect of cell type. This step is essential
# because it adjusts for the fact that measurements within the same cell type
# may be more similar to each other than to measurements from different cell
# types.
corfit <- duplicateCorrelation(v, design, block=block)

# Fit the linear model: These lines fit a linear model to the data, taking into
# account the correlation structure estimated in the previous step, and then
# apply empirical Bayes moderation to the standard errors to improve the
# statistical inference.

# lmFit fits a linear model to the expression data, adjusting for the
# block-specific correlation. This means it models the expression level of each
# gene as a function of the specified experimental conditions (Group) while
# considering the similarity of samples within the same cell type.
fit <- lmFit(v, design, block=block, correlation=corfit$consensus)

# eBayes applies empirical Bayes moderation to the standard errors of the
# estimated coefficients. This step improves the accuracy and stability of the
# differential expression analysis, especially when the number of samples is
# small. It shrinks the standard errors towards a common value, which stabilizes
# the inference of differential expression.

# The fit object contains moderated t-statistics, log-odds of differential
# expression, and other statistics that are adjusted for the correlation within
# blocks and the empirical Bayes moderation.
fit <- eBayes(fit)

# Extract the results: coef: Specifies which coefficient's results to extract.
# "GroupYes" refers to the coefficient for the Group variable, comparing "Yes"
# versus "No". 

# number: Specifies the number of top-ranked genes to return. Inf
# means all genes will be returned.
results <- topTable(fit, coef="GroupYes", number=Inf)
########################

# Extracts results from the DESeq analysis, focusing on the contrast between
# "Cardiomyopathy" and "Control" groups.
deseq_results = results[order(results$adj.P.Val), ]
deseq_results_filtered = subset(deseq_results, P.Value < 0.05)
dim(deseq_results_filtered)
deseq_results$expression = ifelse(
  deseq_results$adj.P.Val < 0.05 &
    (deseq_results$logFC) >= (1.5),
  "Up",
  ifelse(
    deseq_results$adj.P.Val < 0.05 & (deseq_results$logFC) <= (-1.5),
    "Down",
    "Stable"
  )
)
deseq_results$expression = factor(deseq_results$expression, levels = c("Up", "Down", "Stable"))
# deseq_results = subset(deseq_results, abs(log2FoldChange)<5 & !is.na(padj))
deseq_results = subset(deseq_results,!is.na(adj.P.Val))

write.table(deseq_results, file="Pr1_Dox_vs_Control.txt", sep="\t")

#Create Volcano Plot from DESEQ results
p = ggplot(data = deseq_results, aes(x = logFC,
                                     y = -log10(adj.P.Val),
                                     colour = expression,
                                     label = rownames(deseq_results))) +
    geom_point(alpha = 0.4, size = 3.5) +
    scale_color_manual(values=c("red", "blue","grey")) +
    geom_vline(xintercept = c(-2, 2), lty=4, col="black", lwd=0.8) +
    geom_hline(yintercept = -log10(0.05), lty=4, col="black", lwd=0.8) +
    labs(x = expression(bold(log[2]~"fold-change")),
         y = expression(bold(-log[10]~"adjusted p-value"))) +
    theme_bw() +
    theme(panel.grid.major = element_blank(),
          panel.grid.minor = element_blank(),
          text = element_text(size=16, colour = "black", face = "bold"),
          axis.text = element_text(colour = "black", face = "bold"),
          legend.title = element_blank()) +
    scale_x_continuous(breaks = seq(-10, 10,3), limits = c(-10, 10))

tiff(file="Pr1_Dox_vs_Control.tiff",
     width = 18, height = 16, units = 'cm', res = 300, compression = 'lzw')
p
dev.off()
```

# Interpretation:
```{r, eval=TRUE, echo= F}
kable(head(deseq_results), format = "pandoc", caption = "DESeq2 Results (showing first 6 genes only)")
```

**baseMean**: Represents the average of normalized counts across all samples. For example, the gene MEG3 has an average expression level of 17.22.

**log2FoldChange**: Indicates the fold change in expression between the "Cardiomyopathy" and "Control" groups on a logarithmic scale. For instance, MEG3 has a log2 fold change of 21.98, suggesting a substantial increase in expression in the "Cardiomyopathy" group compared to the "Control" group.

**lfcSE (log2FoldChange Standard Error)**: Provides an estimate of the standard error associated with the log2 fold change. A smaller lfcSE indicates a more precise estimate.

**stat**: Represents the Wald statistic, a measure of how many standard errors the estimated log2 fold change is from zero. In this case, higher absolute values of the statistic suggest greater evidence against the null hypothesis.

**pvalue**: Indicates the statistical significance of the log2 fold change. A lower p-value suggests stronger evidence against the null hypothesis.

**padj (adjusted p-value)**: It is the p-value adjusted for false discovery rate (FDR).

**expression**: Categorizes genes based on their expression changes. "Up" indicates upregulation in "Cardiomyopathy," "Down" indicates downregulation, and "Stable" suggests no significant change.


## 2.Pr2

Here, we will design our analysis to identify differentially expressed genes between LVNC and Other_Cardiomyopathy groups, considering the LVNC status as factors in the design formula.We specify the comparison of interest ("LVNC" and "Other_Cardiomyopathy") to compare the gene expression differences associated with the LVNC status.

```{r, eval=TRUE, echo= TRUE}
## remove controls from properties
properties <- read.table(text="Samples	Project	samples.Pr	Cell_type	CO-CULTURED	Dox_treatment
ISO	Pr2	Pr2_ISO	ISO CM	NO	No
ISO_DX	Pr2	Pr2_ISO_DX	ISO CM	NO	Yes
ISO_ISO	Pr2	Pr2_ISO_ISO	ISO CM	ISO CF	No
ISO_ISO_DX	Pr2	Pr2_ISO_ISO_DX	ISO CM	ISO CF	Yes
ISO_R1	Pr2	Pr2_ISO_R1	ISO CM	ROBO1 KO CF	No
ISO_R1_DX	Pr2	Pr2_ISO_R1_DX	ISO CM	ROBO1 KO CF	Yes
ISO_R2	Pr2	Pr2_ISO_R2	ISO CM	ROBO2 KO CF	No
ISO_R2_DX	Pr2	Pr2_ISO_R2_DX	ISO CM	ROBO2 KO CF	Yes
ISO_S2	Pr2	Pr2_ISO_S2	ISO CM	SLIT2 KO CF	No
ISO_S2_DX	Pr2	Pr2_ISO_S2_DX	ISO CM	SLIT2 KO CF	Yes
R1	Pr2	Pr2_R1	ROBO1 KO CM	NO	No
R1_DX	Pr2	Pr2_R1_DX	ROBO1 KO CM	NO	Yes
R1_ISO	Pr2	Pr2_R1_ISO	ROBO1 KO CM	ISO CF	No
R1_ISO_DX	Pr2	Pr2_R1_ISO_DX	ROBO1 KO CM	ISO CF	Yes
R1_R1	Pr2	Pr2_R1_R1	ROBO1 KO CM	ROBO1 KO CF	No
R1_R1_DX	Pr2	Pr2_R1_R1_DX	ROBO1 KO CM	ROBO1 KO CF	Yes
R1_R2	Pr2	Pr2_R1_R2	ROBO1 KO CM	ROBO2 KO CF	No
R1_R2_DX	Pr2	Pr2_R1_R2_DX	ROBO1 KO CM	ROBO2 KO CF	Yes
R1_S2	Pr2	Pr2_R1_S2	ROBO1 KO CM	SLIT2 KO CF	No
R1_S2_DX	Pr2	Pr2_R1_S2_DX	ROBO1 KO CM	SLIT2 KO CF	Yes
R2	Pr2	Pr2_R2	ROBO2 KO CM	NO	No
R2_DX	Pr2	Pr2_R2_DX	ROBO2 KO CM	NO	Yes
R2_ISO	Pr2	Pr2_R2_ISO	ROBO2 KO CM	ISO CF	No
R2_ISO_DX	Pr2	Pr2_R2_ISO_DX	ROBO2 KO CM	ISO CF	Yes
R2_R1	Pr2	Pr2_R2_R1	ROBO2 KO CM	ROBO1 KO CF	No
R2_R1_DX	Pr2	Pr2_R2_R1_DX	ROBO2 KO CM	ROBO1 KO CF	Yes
R2_R2	Pr2	Pr2_R2_R2	ROBO2 KO CM	ROBO2 KO CF	No
R2_R2_DX	Pr2	Pr2_R2_R2_DX	ROBO2 KO CM	ROBO2 KO CF	Yes
R2_S2	Pr2	Pr2_R2_S2	ROBO2 KO CM	SLIT2 KO CF	No
R2_S2_DX	Pr2	Pr2_R2_S2_DX	ROBO2 KO CM	SLIT2 KO CF	Yes
S2	Pr2	Pr2_S2	SLIT2 KO CM	NO	No
S2_DX	Pr2	Pr2_S2_DX	SLIT2 KO CM	NO	Yes
S2_ISO	Pr2	Pr2_S2_ISO	SLIT2 KO CM	ISO CF	No
S2_ISO_DX	Pr2	Pr2_S2_ISO_DX	SLIT2 KO CM	ISO CF	Yes
S2_R1	Pr2	Pr2_S2_R1	SLIT2 KO CM	ROBO1 KO CF	No
S2_R1_DX	Pr2	Pr2_S2_R1_DX	SLIT2 KO CM	ROBO1 KO CF	Yes
S2_R2	Pr2	Pr2_S2_R2	SLIT2 KO CM	ROBO2 KO CF	No
S2_R2_DX	Pr2	Pr2_S2_R2_DX	SLIT2 KO CM	ROBO2 KO CF	Yes
S2_S2	Pr2	Pr2_S2_S2	SLIT2 KO CM	SLIT2 KO CF	No
S2_S2_DX	Pr2	Pr2_S2_S2_DX	SLIT2 KO CM	SLIT2 KO CF	Yes", header = T, sep = "\t")


## data preparaton
counts.matrix_ROBI<-as.matrix(counts2)
mode(counts.matrix_ROBI)<-"integer"

counts.matrix_ROBI <- counts.matrix_ROBI[,grepl("Pr2", colnames(counts.matrix_ROBI))]
colnames(counts.matrix_ROBI) <- gsub("Pr2_", "", colnames(counts.matrix_ROBI))

properties$Cell_type=as.factor(gsub(" ", "_", properties$Cell_type))
properties$tx_Group=as.factor(gsub(" ", "_", properties$Dox_treatment))
properties$culture_Group <- gsub(" ", "_", properties$CO.CULTURED)
properties$sample_ID <- properties$Samples
properties <- properties[, c("sample_ID", "tx_Group", "culture_Group", "Cell_type")] # remove Cell_type
counts.matrix_ROBI <- counts.matrix_ROBI[,match(properties$sample_ID, colnames(counts.matrix_ROBI))]
colnames(counts.matrix_ROBI) == properties$sample_ID


## Create a DESeqDataSet (dds_ROBIS) from the count matrix and phenotype information.
dds_ROBIS<-DESeqDataSetFromMatrix(countData = counts.matrix_ROBI,colData = properties, design=~1)

## Perform variance-stabilizing transformation (vst) on the DESeqDataSet
rld_ROBIS<-vst(dds_ROBIS,blind = FALSE)
mat_ROBIS<-assay(rld_ROBIS)

## Performs principal component analysis (PCA) on the transformed data.
pcaData<-plotPCA(rld_ROBIS, intgroup=c("tx_Group", "Cell_type"), ntop=100, returnData=TRUE)
percentVar<-round(100 * attr(pcaData, "percentVar"))

## Plot PCA
g <- ggplot(pcaData, aes(PC1, PC2))
g <- g + geom_point(size = 5, aes(fill = tx_Group, col = tx_Group, shape = Cell_type)) +
  theme_bw() +
  theme(
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.text = element_text(face = "bold", color = "black"),
    panel.border = element_blank(),
    axis.line = element_line(colour = "black"),
    axis.title = element_text(face = "bold", color = "black"),  # Corrected line
    text = element_text(color = "black", face = "bold")
  )
# Save the plot as a TIFF file
tiff(
  file = "Pr2_top_100_genes.tiff",
  width = 16,
  height = 14,
  units = 'cm',
  res = 300,
  compression = 'lzw'
)

# Print the plot
print(g)

# Close the TIFF device
dev.off()

## Select the top 500 variable genes based on row variances. Create a heatmap
## (phm) using pheatmap with clustering of rows and columns.
library(pheatmap)
properties2<-properties
rownames(properties2)<-properties[,1]
properties2<-subset(properties2, select=-c(sample_ID))
myMatrix<-SummarizedExperiment::assay(rld_ROBIS)
topVarGenes<-head(order(rowVars(myMatrix),decreasing = TRUE), 500)
myMatrixforheatmap<-myMatrix[topVarGenes,]
phm<-pheatmap(myMatrixforheatmap, scale="row", cluster_rows = TRUE, cluster_cols = TRUE, annotation_col = properties2, show_colnames = TRUE, show_rownames = FALSE, legend = T)

g <- phm$gtable
g <- gtable::gtable_add_padding(g, padding = unit(c(l=-2, r=1, t=0, b=0), "cm"))

# Save the heatmap to a file with adjusted dimensions
tiff(file = "Pr2_top_500_genes.tiff", width = 18, height = 32, units = 'cm', res = 600, compression = 'lzw')
grid::grid.draw(g)
dev.off()

# tiff(file="Pr2_top_500_genes.tiff",width = 18, height = 32, units = 'cm', res = 300, compression = 'lzw')
# phm
# dev.off()











properties3 <- properties[properties$Status != "Control", ]

# Create a new column in colData to indicate LVNC status
properties3$LVNC_status <-
  ifelse(properties3$Group=="LVNC",
         "LVNC",
         "Other_Cardiomyopathy")

counts.matrix_ROBI.3 <-
  counts.matrix_ROBI[, match(rownames(properties3), colnames(counts.matrix_ROBI))]

# Run DESeq2 analysis
ddset_lvnc_vs_others <- DESeqDataSetFromMatrix(
  countData = counts.matrix_ROBI.3,
  colData = properties3,
  design = ~ LVNC_status
)

deseq_object_lvnc_vs_others <- DESeq(ddset_lvnc_vs_others)
deseq_results_lvnc_vs_others <-
  data.frame(results(
    deseq_object_lvnc_vs_others,
    contrast = c("LVNC_status", "LVNC", "Other_Cardiomyopathy")
  ))
deseq_results_lvnc_vs_others <-
  deseq_results_lvnc_vs_others[order(deseq_results_lvnc_vs_others$padj), ]
deseq_results_filtered_lvnc_vs_others <-
  subset(deseq_results_lvnc_vs_others, pvalue < 0.05)
dim(deseq_results_filtered_lvnc_vs_others)


deseq_results_lvnc_vs_others$expression = ifelse(
  deseq_results_lvnc_vs_others$padj < 0.05 &
    (deseq_results_lvnc_vs_others$log2FoldChange) >=  (1.5),
  "Up",
  ifelse(
    deseq_results_lvnc_vs_others$padj < 0.05 &
      (deseq_results_lvnc_vs_others$log2FoldChange) <= (-1.5),
    "Down",
    "Stable"
  )
)


deseq_results_lvnc_vs_others$expression = factor(deseq_results_lvnc_vs_others$expression,
                                                 levels = c("Up", "Down", "Stable"))
deseq_results_lvnc_vs_others = subset(deseq_results_lvnc_vs_others,!is.na(padj))


# Write results to a file
write.table(deseq_results_lvnc_vs_others, file = "LVNC_project_deseq_results_LVNC_vs_Other_Cardiomyopathy_Cardiomyocytes.txt", sep = "\t")



# Create Volcano Plot
p_lvnc_vs_others <- ggplot(data = deseq_results_lvnc_vs_others, aes(x = log2FoldChange,
                                                                    y = -log10(padj),
                                                                    colour = expression,
                                                                    label = rownames(deseq_results_lvnc_vs_others))) +
  geom_point(alpha = 0.4, size = 3.5) +
  scale_color_manual(values = c("red", "blue", "grey")) +
  geom_vline(xintercept = c(-2, 2), lty = 4, col = "black", lwd = 0.8) +
  geom_hline(yintercept = -log10(0.05), lty = 4, col = "black", lwd = 0.8) +
  labs(x = expression(bold(log[2] ~ "fold-change")),
       y = expression(bold(-log[10] ~ "adjusted p-value"))) +
  theme_bw() +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        text = element_text(size = 16, colour = "black", face = "bold"),
        axis.text = element_text(colour = "black", face = "bold"),
        legend.title = element_blank()) +
  scale_x_continuous(breaks = seq(-10, 10, 3), limits = c(-10, 10))

# Save the plot as a TIFF file
tiff(file = "Volcano_plot_LVNC_vs_Other_Cardiomyopathy_Cardiomyocytes.tiff",
     width = 18, height = 16, units = 'cm', res = 300, compression = 'lzw')
print(p_lvnc_vs_others)
dev.off()
```

# Interpretation of results:
```{r, eval=TRUE, echo= F}
kable(head(deseq_results_lvnc_vs_others), format = "pandoc", caption = "LVNC and other cardiomyopathy (showing first 6 genesonly)")
```

**baseMean**: Represents the average of normalized counts across all samples. For example, the gene AC069277.1 has an average expression level of 12.35.

**log2FoldChange**: Indicates the fold change in expression between "LVNC" and "Other_Cardiomyopathy" groups on a logarithmic scale. For instance, AC069277.1 has a log2 fold change of -4.79, suggesting a substantial decrease in expression in the "LVNC" group compared to the "Other_Cardiomyopathy" group.

**lfcSE (log2FoldChange Standard Error)**: Provides an estimate of the standard error associated with the log2 fold change. A smaller lfcSE indicates a more precise estimate.

**stat**: Represents the Wald statistic, a measure of how many standard errors the estimated log2 fold change is from zero. In this case, higher absolute values of the statistic suggest greater evidence against the null hypothesis.

**pvalue**: Indicates the statistical significance of the log2 fold change. A lower p-value suggests stronger evidence against the null hypothesis.

**padj (adjusted p-value)**: It is the p-value adjusted for false discovery rate (FDR).

**expression**: Categorizes genes based on their expression changes. "Up" indicates upregulation in "Cardiomyopathy," "Down" indicates downregulation, and "Stable" suggests no significant change.


# Furthermore, we can compare LVNC vs DCM 
```{r, eval=TRUE, echo= TRUE}
# Run DESeq2 analysis
ddset_lvnc_vs_others <- DESeqDataSetFromMatrix(
  countData = counts.matrix_ROBI.3,
  colData = properties3,
  design = ~ Group
)

deseq_object_lvnc_vs_DCM <- DESeq(ddset_lvnc_vs_others)
deseq_results_lvnc_vs_DCM <-
  data.frame(results(
    deseq_object_lvnc_vs_DCM,
    contrast = c("Group", "LVNC", "DCM")
  ))
deseq_results_lvnc_vs_DCM <-
  deseq_results_lvnc_vs_DCM[order(deseq_results_lvnc_vs_DCM$padj), ]
deseq_results_filtered_lvnc_vs_DCM <-
  subset(deseq_results_lvnc_vs_DCM, pvalue < 0.05)
dim(deseq_results_filtered_lvnc_vs_DCM)


deseq_results_lvnc_vs_DCM$expression = ifelse(
  deseq_results_lvnc_vs_DCM$padj < 0.05 &
    (deseq_results_lvnc_vs_DCM$log2FoldChange) >=  (1.5),
  "Up",
  ifelse(
    deseq_results_lvnc_vs_DCM$padj < 0.05 &
      (deseq_results_lvnc_vs_DCM$log2FoldChange) <= (-1.5),
    "Down",
    "Stable"
  )
)


deseq_results_lvnc_vs_DCM$expression = factor(deseq_results_lvnc_vs_DCM$expression,
                                                 levels = c("Up", "Down", "Stable"))
deseq_results_lvnc_vs_DCM = subset(deseq_results_lvnc_vs_DCM,!is.na(padj))


# Write results to a file
write.table(deseq_results_lvnc_vs_DCM, file = "LVNC_project_deseq_results_LVNC_vs_DCM_Cardiomyocytes.txt", sep = "\t")



# Create Volcano Plot
p_lvnc_vs_DCM <- ggplot(data = deseq_results_lvnc_vs_DCM, aes(x = log2FoldChange,
                                                                    y = -log10(padj),
                                                                    colour = expression,
                                                                    label = rownames(deseq_results_lvnc_vs_DCM))) +
  geom_point(alpha = 0.4, size = 3.5) +
  scale_color_manual(values = c("red", "blue", "grey")) +
  geom_vline(xintercept = c(-2, 2), lty = 4, col = "black", lwd = 0.8) +
  geom_hline(yintercept = -log10(0.05), lty = 4, col = "black", lwd = 0.8) +
  labs(x = expression(bold(log[2] ~ "fold-change")),
       y = expression(bold(-log[10] ~ "adjusted p-value"))) +
  theme_bw() +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        text = element_text(size = 16, colour = "black", face = "bold"),
        axis.text = element_text(colour = "black", face = "bold"),
        legend.title = element_blank()) +
  scale_x_continuous(breaks = seq(-10, 10, 3), limits = c(-10, 10))

# Save the plot as a TIFF file
tiff(file = "Volcano_plot_LVNC_vs_DCM_Cardiomyocytes.tiff",
     width = 18, height = 16, units = 'cm', res = 300, compression = 'lzw')
print(p_lvnc_vs_DCM)
dev.off()
```

# Finally, LVNC vs HCM
```{r, eval=TRUE, echo= TRUE}
deseq_object_lvnc_vs_HCM <- DESeq(ddset_lvnc_vs_others)
deseq_results_lvnc_vs_HCM <-
  data.frame(results(
    deseq_object_lvnc_vs_HCM,
    contrast = c("Group", "LVNC", "HCM")
  ))
deseq_results_lvnc_vs_HCM <-
  deseq_results_lvnc_vs_HCM[order(deseq_results_lvnc_vs_HCM$padj), ]
deseq_results_filtered_lvnc_vs_HCM <-
  subset(deseq_results_lvnc_vs_HCM, pvalue < 0.05)
dim(deseq_results_filtered_lvnc_vs_HCM)


deseq_results_lvnc_vs_HCM$expression = ifelse(
  deseq_results_lvnc_vs_HCM$padj < 0.05 &
    (deseq_results_lvnc_vs_HCM$log2FoldChange) >=  (1.5),
  "Up",
  ifelse(
    deseq_results_lvnc_vs_HCM$padj < 0.05 &
      (deseq_results_lvnc_vs_HCM$log2FoldChange) <= (-1.5),
    "Down",
    "Stable"
  )
)


deseq_results_lvnc_vs_HCM$expression = factor(deseq_results_lvnc_vs_HCM$expression,
                                                 levels = c("Up", "Down", "Stable"))
deseq_results_lvnc_vs_HCM = subset(deseq_results_lvnc_vs_HCM,!is.na(padj))


# Write results to a file
write.table(deseq_results_lvnc_vs_HCM, file = "LVNC_project_deseq_results_LVNC_vs_HCM_Cardiomyocytes.txt", sep = "\t")



# Create Volcano Plot
p_lvnc_vs_HCM <- ggplot(data = deseq_results_lvnc_vs_HCM, aes(x = log2FoldChange,
                                                                    y = -log10(padj),
                                                                    colour = expression,
                                                                    label = rownames(deseq_results_lvnc_vs_HCM))) +
  geom_point(alpha = 0.4, size = 3.5) +
  scale_color_manual(values = c("red", "blue", "grey")) +
  geom_vline(xintercept = c(-2, 2), lty = 4, col = "black", lwd = 0.8) +
  geom_hline(yintercept = -log10(0.05), lty = 4, col = "black", lwd = 0.8) +
  labs(x = expression(bold(log[2] ~ "fold-change")),
       y = expression(bold(-log[10] ~ "adjusted p-value"))) +
  theme_bw() +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        text = element_text(size = 16, colour = "black", face = "bold"),
        axis.text = element_text(colour = "black", face = "bold"),
        legend.title = element_blank()) +
  scale_x_continuous(breaks = seq(-10, 10, 3), limits = c(-10, 10))

# Save the plot as a TIFF file
tiff(file = "Volcano_plot_LVNC_vs_HCM_Cardiomyocytes.tiff",
     width = 18, height = 16, units = 'cm', res = 300, compression = 'lzw')
print(p_lvnc_vs_HCM)
dev.off()
```