library(data.table)
setwd("Z:/ResearchHome/Groups/sapkogrp/projects/Genomics/common/POI_genetics/Ruth_et_al")

prsfile <- read.table(text = "SNP	Chr	Pos	Effect_Allele	Other_Allele	Effect_Allele_Freq	Effect	SE	P_value	Direction	HetPVal	N
rs200448	1	 6701978 	T	C	0.4381	0.099	0.012	4.0E-16	+++	6E-01	 200309 
rs9438982	1	 39358143 	C	A	0.3219	0.214	0.013	6.4E-62	---	1E-01	 200308 
rs12046563	1	 43137280 	A	G	0.7609	0.100	0.014	1.7E-12	+++	4E-02	 198947 
rs57259875	1	 45063912 	A	C	0.2551	0.092	0.014	6.6E-11	+++	3E-01	 200308 
rs112869704	1	 46780285 	T	C	0.306	0.158	0.013	2.0E-33	+++	4E-01	 200309 
rs12133213	1	 110191395 	G	A	0.4657	0.080	0.013	5.3E-10	---	5E-01	 200308 
rs11102802	1	 115042415 	T	C	0.2677	0.080	0.014	7.5E-09	+++	8E-01	 200309 
rs72708144	1	 149815740 	C	T	0.0404	0.282	0.032	1.1E-18	---	2E-01	 193479 
rs2736609	1	 156202640 	T	C	0.3592	0.079	0.013	3.0E-10	+++	5E-01	 200309 
rs41272475	1	 169698643 	G	T	0.0053	0.615	0.092	2.4E-11	---	7E-01	 186911 
rs371125724	1	 169897840 	C	CAAAAAAAA	0.1105	0.156	0.023	2.8E-11	?++	4E-01	 140546 
rs11582336	1	 178230945 	A	G	0.7951	0.106	0.015	2.0E-12	+++	4E-01	 200309 
rs1044595	1	 180943529 	T	C	0.5979	0.132	0.012	1.3E-26	+++	3E-01	 200309 
rs7515939	1	 225626327 	A	T	0.3336	0.103	0.013	8.4E-16	+++	4E-01	 201321 
rs4653680	1	 225980267 	A	T	0.4542	0.085	0.012	5.8E-12	+++	2E-01	 201322 
rs7414807	1	 231476874 	A	G	0.6326	0.088	0.013	4.0E-12	+++	6E-01	 199882 
rs1635506	1	 242028804 	C	T	0.4791	0.150	0.012	2.9E-34	---	1E+00	 199883 
rs72755295	1	 242034263 	G	A	0.0323	0.381	0.037	7.9E-25	---	7E-01	 179863 
rs851797	1	 242053042 	G	A	0.8148	0.099	0.016	2.7E-10	---	1E-02	 199882 
rs7539755	1	 244598815 	C	T	0.5766	0.101	0.012	1.4E-16	---	7E-01	 199883 
rs7779	1	 246930564 	C	G	0.0733	0.148	0.024	3.9E-10	+++	7E-01	 199882 
rs112866529	2	 24201599 	G	T	0.8122	0.137	0.016	1.1E-18	---	2E-01	 200310 
rs780088	2	 27716004 	T	C	0.606	0.200	0.012	1.2E-58	+++	9E-01	 199647 
rs12053063	2	 28350611 	G	A	0.7212	0.103	0.014	2.4E-14	---	3E-01	 200309 
rs10198789	2	 28968811 	C	T	0.4156	0.093	0.012	4.5E-14	---	6E-02	 200309 
rs67600408	2	 32793649 	T	C	0.1085	0.140	0.020	1.1E-12	+++	7E-01	 200309 
rs17425341	2	 44324287 	T	G	0.6567	0.085	0.013	2.3E-11	+++	4E-01	 200310 
rs76928871	2	 48005821 	G	A	0.1896	0.198	0.016	2.1E-37	---	6E-01	 200310 
rs2293269	2	 48844968 	G	C	0.6439	0.105	0.013	2.6E-16	---	2E-01	 200310 
rs199638788	2	 61466434 	G	GTCAC	0.714	0.081	0.014	2.6E-09	+++	1E+00	 198475 
rs62156756	2	 67671015 	A	G	0.8963	0.263	0.020	2.2E-40	+++	1E-02	 200309 
rs4852777	2	 71534161 	G	C	0.404	0.074	0.012	2.1E-09	---	7E-02	 200309 
rs67893326	2	 100142349 	A	T	0.7141	0.095	0.014	2.6E-11	+++	3E-01	 200309 
rs59854799	2	 111884592 	C	CT	0.5462	0.082	0.012	2.1E-11	+++	8E-02	 198475 
rs6727266	2	 112007716 	G	A	0.5057	0.079	0.012	9.9E-11	---	7E-01	 200309 
rs72827480	2	 121146501 	C	T	0.4013	0.122	0.013	2.7E-22	---	3E-01	 200309 
rs6430545	2	 135602601 	T	A	0.5732	0.098	0.013	4.9E-15	---	4E-01	 200309 
rs16830019	2	 152267359 	T	G	0.2626	0.118	0.014	2.1E-17	+++	2E-01	 200310 
rs4668354	2	 171814750 	G	C	0.6262	0.187	0.012	5.5E-51	---	9E-01	 200309 
rs201478184	2	 172009538 	CA	C	0.9179	0.220	0.025	6.6E-19	---	9E-01	 198476 
rs6719012	2	 172883647 	T	C	0.426	0.084	0.012	7.0E-12	+++	5E-02	 200309 
rs4972504	2	 177018726 	C	T	0.6926	0.086	0.013	5.3E-11	---	6E-01	 198689 
rs72934556	2	 203990789 	G	T	0.1258	0.101	0.018	3.7E-08	---	5E-01	 200310 
rs540836853	2	 213696913 	T	C	0.9969	0.863	0.148	5.6E-09	?++	6E-01	 140546 
rs6736096	2	 216958369 	C	T	0.5235	0.098	0.012	3.1E-16	---	6E-01	 201322 
rs7558434	2	 223705337 	T	G	0.4388	0.070	0.012	9.1E-09	+++	6E-01	 201322 
rs62193239	2	 230585725 	G	A	0.6612	0.074	0.013	1.0E-08	---	9E-01	 201323 
rs606920	3	 8938965 	A	G	0.8235	0.148	0.016	7.0E-21	+++	7E-01	 200310 
rs12636454	3	 12360214 	C	T	0.2451	0.092	0.014	6.9E-11	---	9E-01	 199072 
rs62244773	3	 30518433 	A	T	0.6987	0.083	0.014	1.8E-09	+++	9E-01	 200309 
rs12487736	3	 47459679 	C	T	0.4205	0.078	0.012	3.5E-10	---	7E-01	 200309 
rs1264191	3	 48649457 	T	C	0.1795	0.108	0.017	7.7E-11	+++	7E-02	 200309 
rs34675186	3	 49765557 	A	G	0.0121	0.480	0.075	1.8E-10	?++	8E-02	 140546 
rs9968117	3	 52838654 	T	C	0.1188	0.106	0.019	1.8E-08	+++	4E-01	 200309 
rs2885255	3	 57722613 	C	G	0.5976	0.071	0.012	8.6E-09	+++	4E-01	 200310 
rs7610102	3	 101246803 	G	A	0.6052	0.077	0.013	1.3E-09	---	6E-01	 200310 
rs4679244	3	 126160529 	A	G	0.5867	0.109	0.012	1.3E-18	+++	3E-01	 200310 
rs6793835	3	 135819934 	G	A	0.7383	0.141	0.014	1.8E-24	---	8E-01	 200309 
rs1545597	3	 142721469 	A	C	0.3617	0.072	0.013	1.5E-08	+++	5E-01	 200309 
rs10154963	3	 150169559 	C	T	0.2798	0.077	0.014	1.8E-08	---	8E-01	 200310 
rs344018	3	 156415993 	G	A	0.5844	0.094	0.012	1.3E-14	---	6E-02	 201323 
rs72268838	3	 183587084 	T	TC	0.2471	0.146	0.014	8.5E-25	+++	5E-01	 199488 
rs10804920	3	 189438689 	T	C	0.4389	0.084	0.012	6.0E-12	+++	1E-01	 201323 
rs979481644	3	 189614828 	A	AT	0.0914	0.126	0.023	2.4E-08	+++	5E-01	 199489 
rs13070791	3	 193385275 	T	C	0.433	0.074	0.012	1.4E-09	+++	9E-03	 201322 
rs3796624	4	 1015844 	G	C	0.2101	0.102	0.015	2.0E-11	---	5E-01	 195967 
rs2052160	4	 13576578 	C	G	0.8928	0.168	0.020	1.9E-17	+++	1E-01	 200310 
rs190043078	4	 25377811 	A	C	0.7764	0.135	0.017	6.3E-16	+++	3E-01	 182729 
rs6824237	4	 38071911 	T	C	0.7506	0.077	0.014	3.5E-08	+++	5E-01	 200310 
rs2581458	4	 41976207 	T	G	0.8137	0.093	0.016	2.5E-09	+++	8E-01	 200310 
rs76540949	4	 48527209 	T	C	0.5221	0.141	0.012	9.6E-31	+++	3E-01	 200309 
rs12651246	4	 84367605 	A	G	0.4875	0.261	0.012	2.2E-103	+++	6E-04	 200310 
rs7698733	4	 95952175 	A	C	0.5441	0.088	0.012	4.9E-13	+++	4E-01	 200309 
rs145615497	4	 99817301 	T	C	0.0161	0.287	0.050	6.6E-09	+++	3E-01	 198667 
rs6810489	4	 99877445 	T	G	0.5958	0.090	0.012	4.4E-13	+++	7E-01	 199073 
rs35713108	4	 101126794 	GA	G	0.6463	0.105	0.013	2.8E-16	---	7E-01	 198475 
rs112898082	4	 155694927 	A	G	0.2603	0.096	0.015	2.8E-10	+?+	8E-01	 166824 
rs2139178	4	 185745029 	T	C	0.3974	0.208	0.012	2.5E-63	+++	2E-02	 201323 
rs9990489	4	 188916240 	C	T	0.5899	0.098	0.013	8.5E-15	---	3E-01	 201323 
rs274701	5	 6728707 	C	A	0.5921	0.144	0.012	3.2E-31	---	5E-01	 200309 
rs80167918	5	 6772341 	G	A	0.9762	0.274	0.041	3.7E-11	---	1E-01	 199926 
rs62356073	5	 36241922 	G	A	0.4264	0.070	0.012	1.6E-08	---	6E-01	 200309 
rs17206591	5	 51997134 	A	C	0.5634	0.074	0.012	1.4E-09	+++	7E-01	 200310 
rs7728833	5	 82000737 	A	G	0.4997	0.079	0.012	5.1E-11	+++	2E-01	 200310 
rs10070308	5	 107281621 	C	T	0.844	0.099	0.017	2.8E-09	---	3E-01	 200309 
rs10477172	5	 141682090 	T	C	0.4883	0.092	0.012	2.8E-14	+++	4E-01	 200310 
rs888694	5	 154307485 	A	C	0.0862	0.160	0.022	1.3E-13	+++	5E-01	 201323 
rs11744866	5	 171934821 	T	C	0.3305	0.103	0.013	2.0E-15	+++	8E-01	 201323 
rs35813302	5	 173396955 	A	G	0.1362	0.236	0.018	3.7E-39	+++	5E-01	 201323 
rs2241584	5	 175956177 	G	A	0.618	0.159	0.013	3.2E-37	---	7E-02	 199960 
rs353478	5	 176370988 	C	T	0.4855	0.318	0.012	4.2E-154	---	9E-03	 201322 
rs76623841	6	 7107404 	G	C	0.0824	0.179	0.024	9.8E-14	---	6E-01	 199893 
rs9348724	6	 10887276 	C	G	0.1724	0.306	0.016	6.9E-80	+++	6E-02	 200309 
rs72823382	6	 10967133 	G	A	0.7389	0.156	0.014	3.9E-29	---	4E-01	 200309 
rs4716056	6	 16278390 	G	A	0.3807	0.076	0.013	1.1E-09	---	3E-01	 200310 
rs9348752	6	 27075483 	C	T	0.1035	0.121	0.020	1.1E-09	---	3E-01	 200185 
rs6930435	6	 29301222 	G	A	0.7524	0.095	0.016	9.1E-10	---	9E-01	 173748 
rs3130040	6	 30528654 	T	A	0.0905	0.202	0.023	1.8E-18	---	1E+00	 176143 
rs4143333	6	 31348519 	A	G	0.8796	0.260	0.022	1.4E-32	+++	7E-02	 176143 
rs2844466	6	 31601012 	T	C	0.6421	0.260	0.013	7.7E-90	+++	2E-01	 198871 
rs9272211	6	 32602029 	T	A	0.9183	0.289	0.033	3.3E-18	?--	9E-01	 140546 
rs9272982	6	 32611376 	G	A	0.7899	0.116	0.020	5.2E-09	?--	1E+00	 140546 
rs7748483	6	 33853071 	A	G	0.9642	0.184	0.033	2.8E-08	+++	5E-01	 197250 
rs9470134	6	 35814325 	C	T	0.3511	0.104	0.013	1.9E-16	---	4E-01	 200309 
rs515650	6	 74099420 	G	T	0.4456	0.083	0.012	1.6E-11	---	7E-01	 200309 
rs113967617	6	 111663858 	A	G	0.1544	0.122	0.017	5.7E-13	+++	5E-01	 200310 
rs2147220	6	 121800148 	C	T	0.3559	0.085	0.013	2.8E-11	---	7E-01	 200309 
rs6569648	6	 130349119 	C	T	0.2358	0.123	0.014	6.9E-18	---	2E-01	 198688 
rs140381358	6	 158575015 	AAAT	A	0.8589	0.115	0.018	1.3E-10	---	9E-01	 199488 
rs67149731	6	 160144783 	CTTTT	C	0.7219	0.091	0.016	1.8E-08	?--	8E-01	 140546 
rs61065130	7	 5444747 	C	CA	0.3776	0.118	0.016	4.7E-14	?++	5E-01	 140546 
rs11767307	7	 23584496 	C	G	0.7697	0.084	0.014	4.9E-09	+++	9E-02	 200309 
rs1826838	7	 45899950 	C	T	0.376	0.088	0.013	3.4E-12	---	1E-01	 200310 
rs62445870	7	 50514904 	T	C	0.0225	0.316	0.043	1.3E-13	+++	8E-01	 199925 
rs375846854	7	 56035325 	A	ATTAT	0.1272	0.149	0.022	1.9E-11	?++	4E-02	 140546 
rs10255049	7	 56121304 	G	A	0.3121	0.140	0.013	1.0E-26	---	9E-01	 200309 
rs2023778	7	 93040470 	G	A	0.2432	0.095	0.014	2.3E-11	---	8E-02	 198688 
rs2056726	7	 99780283 	G	A	0.7803	0.107	0.015	3.4E-13	---	2E-01	 200309 
rs2392836	7	 105973193 	G	A	0.3726	0.115	0.013	4.0E-20	---	3E-01	 200309 
rs4731541	7	 128678236 	C	G	0.3733	0.099	0.013	2.4E-15	+++	5E-01	 201323 
rs112190116	7	 144096158 	T	C	0.0112	0.591	0.071	6.3E-17	?++	9E-02	 140546 
rs2013	7	 158523888 	C	T	0.1659	0.091	0.017	4.9E-08	---	8E-01	 201323 
rs2061834	8	 11452005 	T	G	0.1953	0.108	0.016	1.3E-11	+++	7E-01	 200309 
rs3750243	8	 37888017 	C	G	0.2187	0.386	0.016	3.1E-130	+++	1E-01	 176807 
rs6473979	8	 48957132 	C	A	0.0528	0.213	0.027	3.6E-15	---	3E-01	 200051 
rs3735828	8	 61592425 	G	A	0.3594	0.115	0.013	4.8E-19	---	6E-01	 200310 
rs73264416	8	 81753156 	G	A	0.0771	0.166	0.023	3.3E-13	---	1E-01	 199647 
rs35529153	8	 95631476 	G	A	0.3844	0.074	0.013	3.5E-08	---	3E-01	 200309 
rs117431243	8	 103666218 	A	T	0.951	0.247	0.031	3.6E-15	+++	2E-01	 199582 
rs1467044	8	 120887041 	G	A	0.4597	0.070	0.012	9.7E-09	---	3E-01	 200085 
rs6470598	8	 129018915 	C	T	0.208	0.085	0.015	1.0E-08	---	2E-01	 201323 
rs1476164	8	 129620616 	A	T	0.7837	0.094	0.015	1.3E-10	+++	6E-01	 200660 
rs55873183	9	 878563 	G	A	0.0664	0.191	0.026	4.2E-13	---	6E-01	 198152 
rs783562	9	 23823399 	A	G	0.3698	0.100	0.013	2.6E-15	+++	4E-01	 200309 
rs4879656	9	 33012382 	C	A	0.6248	0.133	0.013	1.9E-26	---	8E-01	 199072 
rs10818873	9	 126559993 	C	T	0.0647	0.218	0.025	1.2E-18	---	7E-01	 199882 
rs138114862	9	 136962817 	C	CA	0.6657	0.080	0.015	5.0E-08	+?+	1E+00	 159383 
rs10795520	10	 5764160 	G	A	0.3884	0.097	0.012	6.7E-15	---	7E-01	 200310 
rs74701710	10	 13208912 	G	A	0.9436	0.216	0.027	1.2E-15	---	9E-01	 198350 
rs10764106	10	 37055384 	C	T	0.5273	0.078	0.012	1.6E-10	---	7E-01	 200310 
rs10997819	10	 69569519 	T	C	0.6579	0.091	0.013	2.2E-12	+++	6E-01	 200310 
rs10823203	10	 70219610 	G	C	0.7924	0.159	0.015	2.9E-26	---	3E-01	 200309 
rs1889921	10	 97803997 	T	G	0.5266	0.089	0.012	1.8E-13	+++	6E-01	 200310 
rs7087644	10	 97826334 	A	G	0.9576	0.289	0.031	3.2E-21	+++	3E-04	 199925 
rs74430332	10	 97958434 	G	A	0.0281	0.281	0.037	5.5E-14	---	1E+00	 199925 
rs10883451	10	 101924418 	C	T	0.4859	0.100	0.012	1.3E-16	---	1E+00	 200310 
rs7091889	10	 104793723 	G	A	0.214	0.121	0.015	2.8E-16	---	7E-01	 200310 
rs11245450	10	 126658075 	G	A	0.5712	0.131	0.012	2.0E-26	---	7E-02	 199700 
rs728900	10	 131590300 	T	A	0.5771	0.144	0.013	6.9E-31	---	1E-01	 201322 
rs6578283	11	 2673575 	G	A	0.2984	0.078	0.014	2.4E-08	---	1E+00	 188327 
rs7928823	11	 9475372 	A	G	0.4285	0.069	0.012	2.2E-08	+++	3E-01	 200310 
rs34591978	11	 16751576 	AATAT	A	0.8055	0.131	0.019	1.9E-12	?--	9E-01	 140546 
rs11031006	11	 30226528 	A	G	0.146	0.201	0.017	1.5E-31	+++	4E-01	 199073 
rs10734411	11	 32541784 	G	A	0.525	0.129	0.012	1.2E-26	---	6E-01	 200309 
rs10769315	11	 48110367 	T	C	0.2089	0.082	0.015	4.3E-08	+++	5E-01	 200310 
rs7943302	11	 76365771 	G	C	0.7923	0.090	0.015	3.1E-09	---	6E-02	 200309 
rs10899493	11	 78117534 	C	T	0.1664	0.120	0.016	1.9E-13	---	9E-01	 200310 
rs11400191	11	 128504173 	T	TA	0.4792	0.076	0.012	7.7E-10	+++	3E-01	 199488 
rs60378595	12	 1004207 	ATAAGT	A	0.7917	0.107	0.016	4.1E-12	---	6E-01	 198476 
rs148681796	12	 6772209 	T	C	0.0141	0.368	0.066	2.0E-08	+?+	7E-01	 154317 
rs10743271	12	 7353790 	T	C	0.7218	0.089	0.014	7.4E-11	+++	9E-01	 200309 
rs2376898	12	 7867200 	T	A	0.7658	0.098	0.015	3.6E-11	---	6E-01	 198413 
rs77100210	12	 10875928 	C	A	0.0508	0.416	0.029	1.2E-46	---	1E-03	 200310 
rs61913600	12	 12884357 	G	A	0.3396	0.087	0.014	1.5E-10	---	5E-02	 200309 
rs10743724	12	 30776022 	C	T	0.5197	0.076	0.012	3.5E-10	---	4E-01	 200309 
rs10219645	12	 48529737 	C	T	0.6863	0.076	0.013	6.9E-09	---	7E-01	 200310 
rs2277339	12	 57146069 	T	G	0.8941	0.370	0.021	4.6E-68	+++	3E-02	 173315 
rs28473132	12	 57707358 	T	A	0.2439	0.084	0.014	3.1E-09	---	2E-01	 200310 
rs7308068	12	 66696706 	C	T	0.4705	0.119	0.012	2.9E-22	---	7E-02	 200310 
rs75770066	12	 66704225 	G	A	0.0319	0.961	0.039	1.8E-137	---	2E-03	 180295 
rs61069255	12	 66816697 	T	A	0.742	0.123	0.014	7.3E-18	---	4E-01	 200310 
rs192638119	12	 66906339 	T	C	0.9775	0.391	0.053	1.3E-13	+++	2E-01	 199925 
rs5800506	12	 102774614 	AT	A	0.5018	0.069	0.012	1.8E-08	---	5E-01	 199489 
rs6490269	12	 120207175 	C	T	0.4823	0.076	0.012	2.6E-10	---	1E-01	 201322 
rs2942371	12	 122131825 	C	T	0.8814	0.179	0.019	1.8E-20	---	2E-01	 201323 
rs73413716	12	 122215128 	G	A	0.0846	0.139	0.022	2.5E-10	---	6E-01	 201323 
rs12825762	12	 122983973 	G	A	0.4495	0.090	0.012	1.9E-13	---	9E-01	 201322 
rs35067339	12	 123760109 	T	A	0.7961	0.178	0.015	7.4E-32	---	3E-01	 201322 
rs28416520	12	 130823657 	G	A	0.5378	0.168	0.012	6.5E-43	---	4E-01	 201323 
rs7318091	13	 31203339 	C	T	0.5954	0.068	0.012	4.1E-08	---	3E-01	 200310 
rs11571815	13	 32968550 	G	A	0.9907	0.561	0.068	1.5E-16	---	5E-01	 177986 
rs112727324	13	 41264694 	T	A	0.3133	0.090	0.016	7.0E-09	?--	5E-01	 140546 
rs3736830	13	 50306221 	C	G	0.8453	0.116	0.017	4.0E-12	+++	2E-02	 200309 
rs7322160	13	 61061456 	C	T	0.3325	0.186	0.013	7.2E-48	---	7E-01	 200310 
rs78566820	13	 70752246 	G	A	0.9928	0.463	0.074	4.1E-10	---	1E+00	 190077 
rs12868295	13	 79986065 	C	T	0.5921	0.098	0.012	2.8E-15	---	5E-01	 200309 
rs112097885	13	 115087096 	T	A	0.7896	0.119	0.018	3.0E-11	?--	3E-01	 140546 
rs1713426	14	 20936077 	A	G	0.6948	0.155	0.014	2.4E-30	+++	5E-01	 200309 
rs12879626	14	 34721134 	T	G	0.3876	0.093	0.013	1.0E-13	+++	6E-01	 198947 
rs61488898	14	 35086437 	C	T	0.9738	0.356	0.040	2.7E-19	---	9E-01	 199925 
rs45547534	14	 45606387 	A	G	0.9822	0.417	0.050	7.3E-17	+++	8E-01	 191229 
rs1969713	14	 45623460 	C	T	0.1044	0.168	0.020	5.3E-17	---	1E+00	 200309 
rs112326803	14	 45825894 	A	G	0.0297	0.254	0.039	9.4E-11	+++	5E-01	 194936 
rs762643	14	 54422767 	G	T	0.5561	0.076	0.012	6.2E-10	---	2E-01	 200309 
rs1986616	14	 73540936 	G	A	0.4773	0.080	0.012	4.3E-11	---	3E-01	 201323 
rs71126035	14	 104019999 	G	GA	0.6958	0.079	0.014	1.5E-08	+++	6E-01	 198050 
rs375453823	15	 41018440 	A	AT	0.546	0.102	0.015	1.2E-11	?++	5E-01	 140546 
rs9796	15	 41271447 	A	T	0.5351	0.181	0.012	1.9E-49	+++	5E-01	 200308 
rs689816	15	 43749852 	A	G	0.2959	0.106	0.013	2.2E-15	+++	6E-01	 200308 
rs11071756	15	 63831636 	C	G	0.8603	0.120	0.017	4.9E-12	+++	6E-01	 201321 
rs8025068	15	 74870045 	G	T	0.8079	0.089	0.016	2.6E-08	---	4E-02	 201322 
rs8035724	15	 78470542 	A	G	0.4261	0.079	0.012	1.6E-10	+++	3E-01	 200085 
rs376442546	15	 83222172 	A	G	0.9948	0.655	0.106	5.3E-10	?++	5E-01	 140546 
rs716886	15	 83715853 	A	G	0.3294	0.084	0.013	7.1E-11	+++	1E+00	 199699 
rs62025270	15	 86300198 	A	G	0.2358	0.093	0.014	8.7E-11	+++	1E-01	 199882 
rs12898357	15	 89801391 	A	G	0.6145	0.183	0.013	1.5E-48	+++	1E-01	 199882 
rs28652789	16	 83807 	G	C	0.7813	0.106	0.016	8.5E-11	---	5E-01	 184260 
rs4786509	16	 4568769 	C	G	0.3043	0.074	0.013	2.8E-08	+++	6E-01	 200309 
rs17680522	16	 9069331 	G	A	0.2926	0.125	0.013	7.9E-21	---	6E-01	 200308 
rs8045589	16	 9210373 	A	T	0.5034	0.100	0.012	1.9E-16	+++	4E-01	 200309 
rs9673473	16	 12011212 	C	G	0.4029	0.212	0.012	3.4E-66	+++	2E-01	 200309 
rs2173885	16	 34992749 	G	A	0.5392	0.133	0.013	9.0E-25	---	8E-01	 200308 
rs11642909	16	 50056698 	A	G	0.6804	0.078	0.013	4.3E-09	+++	2E-01	 200309 
rs11075466	16	 52375540 	G	A	0.1764	0.095	0.016	4.1E-09	---	4E-01	 200308 
rs10521305	16	 53908484 	C	T	0.0597	0.263	0.029	9.6E-20	---	1E+00	 181127 
rs8045027	16	 79388209 	G	A	0.4644	0.073	0.012	2.5E-09	---	8E-01	 197984 
rs200293726	16	 79754440 	T	A	0.3098	0.106	0.016	1.6E-11	?--	8E-01	 140546 
rs12447180	16	 88517722 	C	T	0.3162	0.089	0.013	2.5E-11	---	6E-01	 196463 
rs502258	16	 88968547 	A	G	0.3176	0.083	0.013	4.9E-10	+++	4E-01	 199373 
rs62054610	16	 89816920 	A	G	0.0797	0.176	0.023	1.5E-14	+++	5E-01	 199758 
rs2108839	16	 89860182 	T	G	0.6903	0.136	0.013	6.9E-25	+++	8E-02	 199883 
rs5030755	17	 1782952 	A	G	0.8826	0.128	0.019	2.4E-11	+++	5E-02	 199959 
rs34856659	17	 5327572 	C	T	0.2431	0.168	0.014	7.8E-33	---	9E-01	 201320 
rs222859	17	 7197794 	A	C	0.7815	0.117	0.016	2.2E-13	+++	1E-04	 197504 
rs569145577	17	 18314910 	T	A	0.9933	0.637	0.099	1.5E-10	?--	6E-01	 140546 
rs8073434	17	 35713644 	A	G	0.8086	0.100	0.016	1.8E-10	+++	4E-01	 201321 
rs1565920	17	 37831613 	A	G	0.6811	0.137	0.013	3.9E-26	+++	3E-01	 201320 
rs111637825	17	 40134782 	G	A	0.9357	0.139	0.025	1.9E-08	---	4E-01	 201196 
rs799903	17	 41274906 	A	G	0.3345	0.176	0.013	4.2E-42	+++	3E-01	 201320 
rs1815198	17	 55360585 	A	G	0.3564	0.090	0.013	1.3E-12	+++	6E-01	 199959 
rs1991401	17	 62502435 	A	G	0.6694	0.164	0.014	1.7E-32	+++	7E-03	 193342 
rs34609096	17	 71230680 	A	G	0.4696	0.068	0.012	2.3E-08	+++	1E+00	 201321 
rs4800141	18	 20564275 	A	G	0.6516	0.074	0.013	4.2E-09	+++	4E-01	 200310 
rs142239484	18	 60207797 	T	A	0.9923	0.456	0.080	1.1E-08	---	9E-01	 180577 
rs12605881	18	 60941441 	T	A	0.4055	0.070	0.012	2.0E-08	---	7E-01	 201323 
rs11542373	18	 77663394 	A	G	0.2334	0.129	0.015	3.9E-18	+++	1E+00	 201323 
rs349306	19	 950694 	A	G	0.8693	0.230	0.019	5.3E-33	+++	9E-01	 192964 
rs56225015	19	 3922125 	T	A	0.1319	0.168	0.021	3.1E-15	---	3E-01	 180204 
rs12459191	19	 3938782 	C	T	0.8067	0.120	0.017	9.8E-13	---	4E-01	 192720 
rs147068659	19	 4080518 	T	C	0.0061	1.061	0.106	2.0E-23	?++	9E-02	 140546 
rs148807049	19	 7538568 	C	T	0.882	0.122	0.020	1.1E-09	---	5E-01	 200310 
rs10406053	19	 13987186 	G	C	0.2232	0.102	0.015	2.0E-11	---	7E-01	 198262 
rs191031532	19	 14025178 	C	T	0.9931	0.538	0.093	6.4E-09	?--	8E-01	 140546 
rs2359811	19	 22152876 	G	A	0.1471	0.118	0.017	7.4E-12	---	5E-01	 201323 
rs11670032	19	 23166913 	T	C	0.8557	0.182	0.017	1.6E-25	+++	5E-01	 200085 
rs424223	19	 23485893 	T	G	0.2633	0.100	0.014	4.6E-13	+++	6E-01	 201322 
rs7249357	19	 33465388 	G	A	0.811	0.136	0.016	1.4E-18	---	7E-01	 201322 
rs11349039	19	 46889230 	T	TG	0.7841	0.095	0.015	4.8E-10	+++	3E-01	 199489 
rs10409547	19	 50402819 	G	T	0.3904	0.093	0.015	1.0E-09	?--	7E-01	 140546 
rs11668344	19	 55833664 	A	G	0.6403	0.447	0.013	2.3E-277	+++	2E-01	 201323 
rs117146677	19	 55833868 	G	A	0.9895	0.871	0.068	2.5E-37	---	7E-01	 185328 
rs8109672	19	 56306886 	T	C	0.1625	0.158	0.017	3.5E-21	+++	8E-01	 201323 
rs12461110	19	 56320663 	G	A	0.64	0.168	0.013	1.8E-40	---	1E+00	 199698 
rs8101859	19	 56400161 	G	A	0.2644	0.095	0.014	2.0E-11	---	5E-01	 200086 
rs394448	19	 56452144 	G	C	0.4837	0.079	0.012	6.5E-11	---	7E-01	 201323 
rs6054257	20	 66370 	A	G	0.7951	0.143	0.016	5.7E-20	+++	2E-01	 198411 
rs73078112	20	 5547384 	C	T	0.9757	0.294	0.043	9.5E-12	---	7E-01	 199310 
rs236117	20	 5937175 	G	T	0.8899	0.225	0.019	1.0E-31	---	3E-02	 201322 
rs16991615	20	 5948227 	A	G	0.0625	1.063	0.025	2.34e-387	+++	6E-04	 200085 
rs148563363	20	 5959697 	A	G	0.0057	1.432	0.099	8.9E-48	+++	3E-01	 188830 
rs746748	20	 25282967 	T	C	0.0653	0.219	0.025	7.6E-19	+++	2E-01	 199687 
rs58065489	20	 32052912 	C	G	0.8292	0.098	0.016	1.9E-09	+++	5E-01	 201322 
rs11699793	20	 34271574 	C	T	0.8972	0.123	0.020	7.0E-10	---	4E-01	 200707 
rs483508	20	 48499609 	T	C	0.3847	0.117	0.013	6.8E-21	+++	1E-01	 199690 
rs80047570	20	 49503484 	C	CT	0.7267	0.091	0.015	1.2E-09	+?+	6E-01	 164990 
rs7266248	20	 55183547 	G	A	0.1986	0.115	0.015	3.6E-14	---	4E-01	 201322 
rs73307923	20	 57438940 	A	G	0.0938	0.140	0.022	6.3E-11	+++	2E-01	 201321 
rs6011452	20	 61289445 	C	A	0.785	0.235	0.015	2.0E-54	---	5E-01	 198411 
rs372721829	20	 61294245 	G	A	0.009	0.552	0.079	2.9E-12	?--	8E-01	 140546 
rs10854167	20	 61533039 	G	C	0.7893	0.175	0.015	1.4E-32	---	2E-01	 201321 
rs34978822	20	 62291599 	G	C	0.0192	0.396	0.048	1.4E-16	---	5E-01	 192166 
rs139337637	21	 36449373 	ATTCT	A	0.7047	0.095	0.014	3.2E-12	---	8E-01	 199488 
rs9975728	21	 40662749 	T	G	0.8711	0.104	0.018	7.5E-09	+++	9E-01	 199883 
rs4148974	21	 44323720 	C	T	0.9399	0.160	0.026	4.8E-10	---	6E-01	 198261 
rs5746740	22	 19449473 	G	T	0.8206	0.097	0.016	8.2E-10	---	8E-01	 201322 
rs5754100	22	 21916166 	C	T	0.1906	0.097	0.016	1.9E-09	---	6E-01	 200706 
rs3838149	22	 22298478 	AT	A	0.882	0.142	0.019	1.2E-13	---	2E-01	 191805 
rs62237617	22	 28761148 	T	C	0.0046	1.338	0.120	1.1E-28	+++	7E-08	 182129 
rs5762852	22	 29242473 	C	T	0.1545	0.131	0.017	5.8E-15	---	3E-01	 201321 
rs140353819	22	 29296224 	A	G	0.0102	0.842	0.117	6.4E-13	+++	7E-02	 179598 
rs139906882	22	 29319574 	T	C	0.0125	0.483	0.062	4.2E-15	+++	4E-01	 190931 
rs138430	22	 38874396 	T	C	0.6277	0.182	0.013	4.3E-45	+++	6E-01	 199883 
rs2958663	22	 40394797 	G	C	0.422	0.073	0.012	3.4E-09	---	1E-01	 199883 
rs2272805	22	 45809698 	G	A	0.8545	0.108	0.017	4.3E-10	---	4E-01	 199883 
rs148258173	23	 14839840 	T	C	0.0777	0.221	0.024	1.4E-19	+++	7E-02	 183370 
rs2382759	23	 21874387 	T	C	0.6193	0.130	0.013	2.1E-23	+++	9E-01	 184045 
rs62588628	23	 30753605 	G	A	0.6865	0.204	0.014	5.0E-50	---	5E-02	 184045 
rs5953379	23	 44149649 	C	T	0.9154	0.183	0.023	1.2E-15	---	3E-01	 184045 
rs5981360	23	 74664417 	T	C	0.9629	0.204	0.037	3.0E-08	+++	1E+00	 178641 
rs3213462	23	 152609450 	C	T	0.4977	0.128	0.013	4.5E-24	---	4E-02	 179565", header = T, sep = "\t")

prsfile[] <- lapply(prsfile, function(x) {
  if (is.character(x)) {
    trimws(x)  # Trim whitespace
  } else {
    x  # Return as is if not a character column
  }
})


head(prsfile)

library(biomaRt)
# Use Ensembl SNP dataset (GRCh38)
ensembl <- useEnsembl(biomart = "ENSEMBL_MART_SNP", dataset = "hsapiens_snp", GRCh = 38)
# List of rsIDs
rsids <- prsfile$SNP
# Query to get chromosome and position information
results <- getBM(attributes = c('refsnp_id', 'chr_name', 'chrom_start'),
                 filters = 'snp_filter',
                 values = rsids,
                 mart = ensembl)


# # View the results
print(results)
#
table(results$chr_name)
results <- results[grepl("^[0-9]+$|X|Y", results$chr_name),]
results

prsfile$POS_GRCh38 <- results$chrom_start[match(prsfile$SNP, results$refsnp_id)]

# I checked the ones from UCSC table browser and they look good.
# prsfile$bed37 <- paste0("chr", prsfile$Chr, " ", prsfile$Pos, " ", prsfile$Pos)
# setwd("Z:/ResearchHome/Groups/sapkogrp/projects/Genomics/common/POI_genetics/Ruth_et_al")
# write.table(as.data.frame(prsfile$bed37), "prsfile_liftover_37.txt", col.names = F, row.names = F, quote = F)
# 
# 
# # Based on UCSC liftover, these were deleted in new GRCh38
# # #Deleted in new
# # chr23:14839840-14839840
# # #Deleted in new
# # chr23:21874387-21874387
# # #Deleted in new
# # chr23:30753605-30753605
# # #Deleted in new
# # chr23:44149649-44149649
# # #Deleted in new
# # chr23:74664417-74664417
# # #Deleted in new
# # chr23:152609450-152609450
# 
# ## Read file from UCSC browser
# ucsc.df <- read.delim("hgTables_GRCh38.txt", sep = "\t")
# ucsc.df <- ucsc.df[!grepl("fix|alt", ucsc.df$chrom),]
# 
# prsfile$SNP[!(prsfile$SNP %in% ucsc.df$name)]
# prsfile$matchrsid <- prsfile$SNP %in% ucsc.df$name
# prsfile$matchGRCh38 <- prsfile$POS_GRCh38 %in% ucsc.df$chromEnd
# prsfile$UCSC.GRCh38 <- ucsc.df$chromEnd[match(prsfile$SNP, ucsc.df$name)]

filtered_prsfile <- prsfile[
  nchar(prsfile$Effect_Allele) > 1 | nchar(prsfile$Other_Allele) > 1, 
]

## So basically, we need to substract 1 base from indel sites 
prsfile$POS_GRCh38[
  nchar(prsfile$Effect_Allele) > 1 | nchar(prsfile$Other_Allele) > 1
] <- prsfile$POS_GRCh38[
  nchar(prsfile$Effect_Allele) > 1 | nchar(prsfile$Other_Allele) > 1
] -1


prsfile$CHROM <- paste0("chr", prsfile$Chr)
prsfile$CHROM[prsfile$CHROM == "chr23"] <- "chrX"
prsfile$CHROM[prsfile$CHROM == "chr24"] <- "chrY"
#####################
## GRCh38 PRS file ##
#####################
## should be in this format
# chr1 17445652 G A 0.156747305674524 ST6 ST6
# chr1 37887731 T C 0.0441608957857698 ST6 ST6
# chr1 41446467 T C 0.0777940274400202 ST6 ST6
POI_meta_GRCh38 <- cbind.data.frame(prsfile$CHROM, prsfile$POS_GRCh38, prsfile$Other_Allele, prsfile$Effect_Allele, prsfile$Effect, "POI_META", "POI_META")

# Note: No need to swap the allele based on "Direction +++ or ---" from meta analysis. You just need to make sure beta is positive. If you change its sign, you will need to swap the alleles
write.table(POI_meta_GRCh38, "POI_meta_GRCh38.dat", col.names = F, row.names = F, quote = F)

# extract from BIM
POI_meta_GRCh38$KEY <- paste0(POI_meta_GRCh38$`prsfile$CHROM`, ":", POI_meta_GRCh38$`prsfile$POS_GRCh38`)

wantedSNP <- c()
# Loop through chromosomes 1-22, X, Y
for (CHR in c(1:22, "X", "Y")) {
  print(paste0("Doing chr", CHR))
  # Construct the file path for each chromosome's BIM file
  BIMfile <- paste0("Survivor_WGS.GATK4180.hg38_renamed_chr", CHR, ".PASS.decomposed.qced.bim")
  BIMfile <- paste0("Z:/ResearchHome/Groups/sapkogrp/projects/Genomics/common/Survivor_WGS_QCed/QC/", BIMfile)

  # Read the BIM file
  df <- fread(BIMfile)
  
  # Create the KEY for each row by combining chromosome and position
  df$KEY <- paste0("chr", df$V1, ":", df$V4)
  # Find SNPs in the POI_meta_GRCh38 based on matching KEYs
  tmp.wantedSNP <- df$V2[df$KEY %in% POI_meta_GRCh38$KEY]
  # Append to the wantedSNP list
  wantedSNP <- c(wantedSNP, tmp.wantedSNP)
}

wantedSNP_df <- data.frame(SNP = wantedSNP)
write.table(wantedSNP_df, "wantedSNP_GRCh38.txt", row.names = FALSE, col.names = F, sep = "\t", quote = FALSE)


bimcheck <- fread(".//prs/prs_out/POI_META.bim")
bimcheck$KEY <- paste0("chr", bimcheck$V1, ":", bimcheck$V4)

POI_meta_GRCh38[!POI_meta_GRCh38$KEY %in% bimcheck$KEY,]

#####################
## GRCh37 PRS file ##
#####################
## should be in this format
# chr1 17445652 G A 0.156747305674524 ST6 ST6
# chr1 37887731 T C 0.0441608957857698 ST6 ST6
# chr1 41446467 T C 0.0777940274400202 ST6 ST6
POI_meta_GRCh37 <- cbind.data.frame(prsfile$CHROM, prsfile$Pos, prsfile$Other_Allele, prsfile$Effect_Allele, prsfile$Effect, "POI_META", "POI_META")

write.table(prsfile, "POI_meta_GRCh37.dat", col.names = T, row.names = F, quote = F)


# extract from BIM
POI_meta_GRCh37$KEY <- paste0(POI_meta_GRCh37$`prsfile$CHROM`, ":", POI_meta_GRCh37$`prsfile$Pos`)

wantedSNP <- c()
# Loop through chromosomes 1-22, X, Y
for (CHR in c(1:22, "X", "Y")) {
  print(paste0("Doing chr", CHR))
  # Construct the file path for each chromosome's BIM file
  BIMfile <- paste0("CCSS_org_GRCh37_chr", CHR, ".bim")
  BIMfile <- paste0("Z:/ResearchHome/Groups/sapkogrp/projects/Genomics//common/ccss_org_hrc/plink/", BIMfile)
  
  # Read the BIM file
  df <- fread(BIMfile)
  
  # Create the KEY for each row by combining chromosome and position
  df$KEY <- paste0(df$V1, ":", df$V4)
  # Find SNPs in the POI_meta_GRCh37 based on matching KEYs
  tmp.wantedSNP <- df$V2[df$KEY %in% POI_meta_GRCh37$KEY]
  # Append to the wantedSNP list
  wantedSNP <- c(wantedSNP, tmp.wantedSNP)
}

wantedSNP_df <- data.frame(SNP = wantedSNP)
write.table(wantedSNP_df, "wantedSNP_GRCh37.txt", row.names = FALSE, col.names = F, sep = "\t", quote = FALSE)
